packages/mongodb-chatbot-ui/src/Message.tsx

```diff
@@ -317,33 +317,37 @@ export const MessagePrompts = ({
   messagePrompts,
   messagePromptsOnClick,
 }: MessagePromptsProps) => {
-  const [inProp, setInProp] = useState(true);
+  const [showPrompts, setShowPrompts] = useState(true);
   const [suggestedPromptIdx, setSuggestedPromptIdx] = useState(-1);
   const nodeRef = useRef(null);
   const duration = 300;
 
+  const onPromptSelected = (prompt: string, idx: number) => {
+    setSuggestedPromptIdx(idx);
+    setShowPrompts(false);
+    setTimeout(() => {
+      messagePromptsOnClick(prompt);
+    }, duration);
+  };
+
   return (
     <CSSTransition
-      in={inProp}
+      in={showPrompts}
       timeout={duration}
       nodeRef={nodeRef}
       classNames={styles.message_prompts}
     >
       <div className={styles.message_prompts} ref={nodeRef}>
         <LGMessagePrompts label="Suggested Prompts">
-          {messagePrompts.map((sp, idx) => (
+          {messagePrompts.map((suggestedPrompt, idx) => (
             <MessagePrompt
-              key={sp}
-              onClick={() => {
-                setSuggestedPromptIdx(idx);
-                setInProp(false);
-                setTimeout(() => {
-                  messagePromptsOnClick(messagePrompts[idx]);
-                }, duration);
-              }}
+              key={suggestedPrompt}
+              onClick={() =>
+                showPrompts === true && onPromptSelected(suggestedPrompt, idx)
```

Comment:
Maybe could use a comment here so we remember later why we did this. Also may as well put the throttling functionality in `onPromptSelected()` so we don't have logic spread across two places.

Reply:
I'm mildly  concerned about `showPrompts` as a throttling check variable - since it's a state variable it will take a whole render before it is actually set to false. So any clicks between first click and re-render (however small a window) will still register with `showPrompts` still true?

Reply:
1. Yeah we can move the check to the implementation of `onPromptSelected()`
2. I _think_  React handles this pretty gracefully but I could misunderstand the rendering/batching process. I think I can change to use a ref value that works independently of the render cycle. Lemme look into it.